sssd_test_framework/roles/keycloak.py

```diff
@@ -152,6 +209,8 @@ def get(self) -> dict[str, list[str]]:
             out.setdefault(key, [])
             out[key].append(json1[key])
 
+        self.id = out["id"][0]
```

Comment:
Please, don't create fields outside of the constructor, it is a bad habit. If you do this, you never know what properties are available (because they can be set anywhere) and you can't document them. Instead, add this line to the constructor.

Reply:
Yes, it looked strange and I proposed returning a tuple like password authentication, and then Scott said he didn't need it after all.

Reply:
@danlavu I think I was confused over the point you were making earlier..

@pbrezina isn't self.id inherited from the KeycloakObject parent class?

Reply:
> @danlavu I think I was confused over the point you were making earlier..
> 
> @pbrezina isn't self.id inherited from the KeycloakObject parent class?

Ah, yes, you are right. It wasn't clear from the diff. In that case, you still should put it to the User class because it is not used anywhere else, right?

Reply:
Several Keycloak commands require ID so it's also used in the KeycloakGroup class.  Should I move it from KeycloakObject into each class where it's needed?   Right now KeycloakObject only defines `name` and `id`.

Reply:
Ok, I went through the whole file and see what you are doing now. Keep it as is, but maybe make it little bit safer to use to avoid hunting potential bugs:

```
class KeycloakObject(BaseObject[KeycloakHost, Keycloak]):
    """
    Keycloak Object.
    """

    def __init__(self, role: Keycloak, name: str) -> None:
        """
        :param role: Keycloak role object.
        :type role: Keycloak
        :param name: User name.
        :type name: str
        """
        super().__init__(role)
        self.name = name

        # This is set as a side effect of certain operations like add or get.
        self._id: str | None = None

    @property
    def id(self) -> str:
        """
        Keycloak object ID.

        :raises ValueError: If ID is not set.
        :return: Keycloak object ID.
        :rtype: str
        """
        if self._id is None:
            raise ValueError("Keycloak object ID is unset.")

        return self._id

    @id.setter
    def id(self, value: str):
        self._id = value
```

Reply:
Forgot to mention earlier that I made the suggested change.